search/management/commands/recreate_index.py

```diff
@@ -1,50 +1,19 @@
 """
 Management command to recreate the Elasticsearch index
 """
-import logging
 
 from django.core.management.base import BaseCommand
-
-from search.indexing_api import (
-    recreate_index,
-    __name__ as indexing_api_name,
-)
+from search.tasks import recreate_index_async
 
 
 class Command(BaseCommand):
     """
     Command for recreate_index
     """
-    help = "Clears existing Elasticsearch indices and creates a new index and mapping."
-
-    def add_arguments(self, parser):  # pylint: disable=no-self-use
-        """Configure command args"""
-        parser.add_argument(
-            '--profile',
-            action='store_true',
-            dest='profile',
-            default=False,
-        )
+    help = "Starts a new celery task that clears existing Elasticsearch indices and creates a new index and mapping."
 
     def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
         """
         Recreates the index
         """
-        log = logging.getLogger(indexing_api_name)
-        console = logging.StreamHandler(self.stderr)
-        console.setLevel(logging.DEBUG)
-        log.addHandler(console)
-        log.level = logging.INFO
-
-        if kwargs['profile']:
-            import cProfile
-            import uuid
-            profile = cProfile.Profile()
-            profile.enable()
-            recreate_index()
-            profile.disable()
-            filename = 'recreate_index_{}.profile'.format(uuid.uuid4())
-            profile.dump_stats(filename)
-            self.stdout.write('Output profiling data to: {}'.format(filename))
-        else:
-            recreate_index()
+        recreate_index_async.delay()
```

Comment:
Can we write out the task id to the console here?

Reply:
@noisecapella, Thanks for pointing this out I was writing the task id within the task but the command will be a better place for it. I've now updated the PR as per your suggestion and this is ready for review.

Also, Just FYI, There is another [PR](https://github.com/mitodl/micromasters/pull/4968) that is actually rebased on this one. That PR is the successor part of what we wanted to achieve in the ticket. The associated ticket has more details regarding that. You can take a look at that as well and we can decide either to merge this PR and check out the feasibility of the other one or close this PR in favor of the other [PR](https://github.com/mitodl/micromasters/pull/4968) which is rebased on this.

Reply:
I think it makes sense to merge this and then continue the work in #4968